src/theory/datatypes/sygus_datatype_utils.cpp

```diff
@@ -605,6 +606,40 @@ Node getExpandedDefinitionForm(Node op)
   return eop.isNull() ? op : eop;
 }
 
+void computeExpandedDefinitionForms(Env& env, const TypeNode& tn)
+{
+  std::unordered_set<TypeNode> processed;
+  std::vector<TypeNode> toProcess;
+  toProcess.push_back(tn);
+  size_t index = 0;
+  while (index < toProcess.size())
+  {
+    TypeNode tnp = toProcess[index];
+    index++;
```

Comment:
I think it's easier to follow a stack-like behavior.
```suggestion
  while (!toProcess.empty())
  {
    TypeNode tnp = toProcess.back();
    toProcess.pop_back();
```